controlplane/kubeadm/internal/etcd/etcd.go

```diff
@@ -234,3 +238,17 @@ func (c *Client) Alarms(ctx context.Context) ([]MemberAlarm, error) {
 
 	return memberAlarms, nil
 }
+
+// HealthCheck checks the healthiness of endpoints specified in endpoints during Client creation.
+// Using the same logic used in etcdctl health command.
+func (c *Client) HealthCheck(ctx context.Context) error {
+	// This check is return nil if FakeEtcdClient.
+	if c.KVClient == nil {
+		return nil
+	}
```

Comment:
Coding up logic for testing can usually impact real environment as well. Can we satisfy the interface with a different client in this case?

Reply:
Yes, implemented fake KV interface.